core/src/nodes/collection.rs

```diff
@@ -439,6 +439,17 @@ impl<TInEvent, TOutEvent, THandler> CollectionStream<TInEvent, TOutEvent, THandl
     }
 }
 
+/// Errors for interrupt(). 
+#[derive(Debug)]
+pub enum InterruptErr {
```

Comment:
It'd be good if this implemented a few traits, [`std::error::Error`](https://doc.rust-lang.org/std/error/trait.Error.html) and [`std::fmt::Display`](https://doc.rust-lang.org/std/fmt/trait.Display.html) so it becomes easier to use elsewhere in the code. Usually Rust error types are named `*Error` (not just `*Err`).